### ISSUE TITLE
Add extreme precip layers

### ISSUE THREAD

**Issue by sasignell** [caryfitzhugh/NYCCSC-map-viewer#138]
Extreme precip layers
 http://maps.naturalresourcenavigator.org/ 
These are in the climate tab.
An An example info request
The ArcGIS REST Url

**Comment by sasignell** [comment]
Which layers-- the 10 year, 100?  The 100 ones are pretty course scale, and the 10 year ones don't show much interesting changes.

**Comment by colinbeier** [comment]
are 10 and 100 years the only options?  the data is quite coarse, it is produced out of Art's shop, NYSERDA asked to see it in there.

**Comment by sasignell** [comment]
These are the layers I'm looking at:
[image: Inline image 1]

On Thu, Feb 11, 2016 at 11:12 AM, colinbeier <notifications@github.com>
wrote:

> are 10 and 100 years the only options? the data is quite coarse, it is
> produced out of Art's shop, NYSERDA asked to see it in there.
>
> On Feb 11, 2016, at 11:10 AM, Steve Signell wrote:
>
> > Which layers-- the 10 year, 100? The 100 ones are pretty course scale,
> and the 10 year ones don't show much interesting changes.
> >
> > —
> > Reply to this email directly or view it on GitHub.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caryfitzhugh/NYCCSC-map-viewer/issues/138#issuecomment-182938597>
> .
>



-- 
Steve Signell
Frontier Spatial, L.L.C.
1062 Dean St.
Schenectady, NY 12309
518-232-1008
frontierspatial.com
steve@frontierspatial.com

**Comment by sasignell** [comment]
Hey Colin, can you go to http://maps.naturalresourcenavigator.org/ and take a look at the extreme precip layers and then tell me what to bring in?  I'm not really sure what I'm looking at there.

**Comment by colinbeier** [comment]
There are 6 layers, I think we can collapse them down into a single layer with toggles.  The variables are 1) return interval (10 yr, 100 yr), 2) time period (current, future % change, future return interval).   The 4 future layers are coarse grids, while the 2 current layers are much higher-resolution maps.   I don't know if this creates an issue.  

As with other climate layers, it would be helpful to be able to click on the map and return a value.

**Comment by sasignell** [comment]
I'll give this a go this morning

**Comment by caryfitzhugh** [comment]
@sasignell how'd this go? I've got a "method" now, since the rare plants was pulled in, so I can do this one pretty easily if you want.

**Comment by sasignell** [comment]
working on it right now... give me a few minutes to see if I can figure it
out.

On Wed, Mar 2, 2016 at 3:36 PM, Cary FitzHugh <notifications@github.com>
wrote:

> @sasignell <https://github.com/sasignell> how'd this go? I've got a
> "method" now, since the rare plants was pulled in, so I can do this one
> pretty easily if you want.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caryfitzhugh/NYCCSC-map-viewer/issues/138#issuecomment-191418377>
> .
>



-- 
Steve Signell
Frontier Spatial, L.L.C.
1062 Dean St.
Schenectady, NY 12309
518-232-1008
frontierspatial.com
steve@frontierspatial.com

**Comment by caryfitzhugh** [comment]
cool. I was able to parse the legend from their service. 
And once that was done, it was just a matter of mapping the selection (dropdowns) to a layer_id.

**Comment by sasignell** [comment]
I think I got it.   Colin & Cary-- check it out to see if it's ok

**Comment by sasignell** [comment]
http://52.2.5.122/NYCCSC-map-viewer/src/map_viewer.html?active-layers=extreme_precip

**Comment by colinbeier** [comment]
Looks great!  Can we make it clickable to return a value?  

Also, I see this layer in the tree on the gh pages, but it just spins, does not load.  just FYI.

**Comment by sasignell** [comment]
Yeah I can add that I think.


Also, Colin-- can you come up with some text for the description, or should
we let Ingrid do that?

On Wed, Mar 2, 2016 at 11:42 PM, colinbeier <notifications@github.com>
wrote:

> Looks great! Can we make it clickable to return a value?
>
> Also, I see this layer in the tree on the gh pages, but it just spins,
> does not load. just FYI.
>
> On Mar 2, 2016, at 4:18 PM, Steve Signell wrote:
>
> >
> http://52.2.5.122/NYCCSC-map-viewer/src/map_viewer.html?active-layers=extreme_precip
>
> >
> > —
> > Reply to this email directly or view it on GitHub.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caryfitzhugh/NYCCSC-map-viewer/issues/138#issuecomment-191578772>
> .
>



-- 
Steve Signell
Frontier Spatial, L.L.C.
1062 Dean St.
Schenectady, NY 12309
518-232-1008
frontierspatial.com
steve@frontierspatial.com

**Comment by colinbeier** [comment]
a few issues with the layer on Steve's dev site:  

1. the legend is funky, has three parts of a color bar instead of a complete bar. most of the boxes have a color gradient bottom to top, which is weird.  not sure what's going on there.  

2. the units need to be added and they change depending on the data that's being shown. for the timeframe=current magnitude, the units should be converted to inches from the raw data, which is in .001 inches.  for timeframe=future percent increase the units should be percentage points (relative to present day).  for timeframe=future recurrence the units should be years.  

3. on Steve's dev site the legend color bar does not reset back once you toggle away from the initial setting.  so if i go from current magnitude to future recurrence, the color bar changes, but if i toggle back to current magnitude it does not change back again.  

4. replace 'Species' with 'Storm Type', Recurrence is misspelled in the toggle

5. can we add the GetInfo function to this one?

**Comment by sasignell** [comment]
The legend is the same layout as the rare plants one-- I basically used the
same template.

A bunch of issues here-- I don't have time to address this afternoon--
heading out shortly for the ADKs.  I vote for leaving it off this release.

On Fri, Mar 4, 2016 at 1:07 PM, colinbeier <notifications@github.com> wrote:

> a few issues with the layer on Steve's dev site:
>
>    1.
>
>    the legend is funky, has three parts of a color bar instead of a
>    complete bar. most of the boxes have a color gradient bottom to top, which
>    is weird. not sure what's going on there.
>    2.
>
>    the units need to be added and they change depending on the data
>    that's being shown. for the timeframe=current magnitude, the units should
>    be converted to inches from the raw data, which is in .001 inches. for
>    timeframe=future percent increase the units should be percentage points
>    (relative to present day). for timeframe=future recurrence the units should
>    be years.
>    3.
>
>    on Steve's dev site the legend color bar does not reset back once you
>    toggle away from the initial setting. so if i go from current magnitude to
>    future recurrence, the color bar changes, but if i toggle back to current
>    magnitude it does not change back again.
>    4.
>
>    replace 'Species' with 'Storm Type', Recurrence is misspelled in the
>    toggle
>    5.
>
>    can we add the GetInfo function to this one?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caryfitzhugh/NYCCSC-map-viewer/issues/138#issuecomment-192386668>
> .
>



-- 
Steve Signell
Frontier Spatial, L.L.C.
1062 Dean St.
Schenectady, NY 12309
518-232-1008
frontierspatial.com
steve@frontierspatial.com

**Comment by colinbeier** [comment]
ok, well rare plants and animals is not working for me either (just spins), so I cannot compare them, I'm not sure what's up.  

we can leave it off this release.

**Comment by caryfitzhugh** [comment]
The legends will look like that since we're pulling the legend info
straight from the WMS server.  ANd they only give us 3 legend blocks, with
image colors, so we can't even really gradient-ize it ourselves.  I guess
maybe we could with canvas, but...   bleck.

I'll have to still look at that CDN issue.

**Comment by colinbeier** [comment]
OK good to know.

About the CDN, if they rejecting it somehow on their end, it's a NYSERDA funded site so I can make a couple of calls and get that sorted...

Colin Beier
SUNY ESF

On Mar 4, 2016 2:02 PM, Cary FitzHugh <notifications@github.com> wrote:
The legends will look like that since we're pulling the legend info
straight from the WMS server. ANd they only give us 3 legend blocks, with
image colors, so we can't even really gradient-ize it ourselves. I guess
maybe we could with canvas, but... bleck.

I'll have to still look at that CDN issue.

**Comment by caryfitzhugh** [comment]
* Won't fix funky legend.

* **Units**  This is tricky, I don't think we can easily change this.  This is their legend information.  I am hesitant to do much twiddling, b/c if they change anything, we break. 
  I would recommend contacting them and saying they should make the changes on their map server.
If they promote the code from
http://dev.services2.coastalresilience.org/arcgis/rest/services/New_York/NY_CLIMAD_toolkit/MapServer
to 
http://services.coastalresilience.org/arcgis/rest/services/New_York/NY_CLIMAD_toolkit/MapServer
The development one has slightly better formatted labels.  Still not great, but at least it has " on some.  DOesn't have % or years in the labels though.

* legend color bar --- Fixed. Found.

* **can we add the GetInfo function to this one?**
  I think we could work out the details (my quick examination of this:
http://dev.services2.coastalresilience.org/arcgis/rest/services/New_York/NY_CLIMAD_toolkit/MapServer/identify

 means it is a task for another day), but this is all you get - not all that great... just a number.
![image](https://cloud.githubusercontent.com/assets/445848/13538478/bebed11a-e218-11e5-9ad6-c59e1310c633.png)

**Comment by caryfitzhugh** [comment]
Summary:

**Won't Fix** Funky Legend
**Did Fix** reloading legend gets updated
**Proposed Alternative** Have the map people update their legends on their end.
**TBD** GetFeatureInfo calls for the rare plants and extreme precip layers.

**Comment by colinbeier** [comment]
Got it, thanks!  

TLDR: word.

**Comment by caryfitzhugh** [comment]
@colinbeier I have figured out how to make the identify calls to esri map servers (power stations).

Did you want this info from extreme precip layers?
I guess maybe rare plants as well... ?

**Comment by colinbeier** [comment]
Sure, if it's easy peasy... would be great!

Colin Beier
SUNY ESF

On Mar 24, 2016 4:07 PM, Cary FitzHugh <notifications@github.com> wrote:

@colinbeier<https://github.com/colinbeier> I have figured out how to make the identify calls to esri map servers (power stations).

Did you want this info from extreme precip layers?
I guess maybe rare plants as well... ?

**Comment by caryfitzhugh** [comment]
rare plants and animals had it already.
extreme precip now does it.

**Comment by colinbeier** [comment]
Nice!

Colin Beier
SUNY ESF

On Mar 25, 2016 9:58 AM, Cary FitzHugh <notifications@github.com> wrote:

rare plants and animals had it already.
extreme precip now does it.
